flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -742,6 +755,24 @@ public FileVisitResult preVisitDirectory(java.nio.file.Path dir, BasicFileAttrib
 			}
 		}
 
+		for (URI originURI : archives) {
+			Path remoteParent = Utils.getRemoteResourceRoot(appId.toString(), new Path(originURI.getPath()), fs.getHomeDirectory());
+			String fragment = originURI.getFragment();
+			Path target = new Path(
+				FileUtils.localizeRemoteFiles(new org.apache.flink.core.fs.Path(remoteParent.toUri()), originURI).toUri());
```

Comment:
Maybe `FileUtils.localizeRemoteFiles` should return a `URI` if the returned `Path` has to be unpacked and then again packed into another `Path` type.